Source/WebCore/html/HTMLMediaElement.cpp

```diff
@@ -7521,6 +7521,9 @@ void HTMLMediaElement::createMediaPlayer() WTF_IGNORES_THREAD_SAFETY_ANALYSIS
     player->setPageIsVisible(!m_elementIsHidden, page ? page->sceneIdentifier() : ""_s);
     player->setVisibleInViewport(isVisibleInViewport());
     schedulePlaybackControlsManagerUpdate();
+#if ENABLE(LEGACY_ENCRYPTED_MEDIA) && ENABLE(ENCRYPTED_MEDIA)
+    updateShouldContinueAfterNeedKey();
```

Comment:
I would place this in mediaPlayerEngineUpdated as it gets updated when we get a new MediaPlayerPrivate and you wouldn't need to cache it all like you've done.

I would make MediaPlayer::setShouldContinueAfterKeyNeeded no longer be virtual, but store the value ; so that any MediaPlayerPrivate querying its MediaPlayer would get it.